### PR TITLE
adds release guild members, and auto generated guild membership

### PR DESCRIPTION
**Pull request by coury-clark**
Adds a new type of membership called `guild`. This can autopopulate a list of members from team references.
CleanShot 2022-02-16 at 12 49 19@2x

### CODE REVIEW

**Comment by michaellzc** [comment]
so many CI checks 😂

looks like we need a `guilds.schema.json` defined for the new yaml

https://github.com/sourcegraph/handbook/runs/5222333622?check_suite_focus=true#step:6:11

**Comment by coury-clark** [comment]
@michaellzc  

ah yeah - just added that

File: data/guilds.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+release_guild:
+  title: Release Guild
+  leader: coury_clark
+  member_refs: [coury_clark, joe_chen, camden_cheek, tomás_senart, keegan_carruthers-smith, mohammad_umer_alam]
```
Comment (jasonyavorsky) [inline data/guilds.yml]: Maybe just call this `members`?

**Comment by jasonyavorsky** [comment]
Since you're adding references, you should also update https://github.com/sourcegraph/handbook/blob/main/src/scripts/validateData.mjs to make sure that the refs you point to exist.

See https://handbook.sourcegraph.com/handbook/editing/content-generation/#code for more info.

**Review by jasonyavorsky — APPROVED** [review]

**Comment by coury-clark** [comment]
> Since you're adding references, you should also update https://github.com/sourcegraph/handbook/blob/main/src/scripts/validateData.mjs to make sure that the refs you point to exist.
> 
> See https://handbook.sourcegraph.com/handbook/editing/content-generation/#code for more info.

Added!

**Comment by jasonyavorsky** [comment]
Nice job @coury-clark !

**Pull request merged**